src/game/ingame_menu.c

```diff
@@ -308,35 +327,40 @@ static s32 is_color_code_valid(char *str, s32 strPos) {
  * Get the exact width of the line of a string of any font in pixels, using the given ASCII and UTF-8 tables.
  */
 s32 get_string_width(char *str, struct AsciiCharLUTEntry *asciiLut, struct Utf8LUT *utf8LUT) {
+    char c;
     s32 width = 0;
     s32 maxWidth = 0;
-    s32 strPos = 0;
-    char c;
+    s32 strPos = -1;
 
     s32 isGenericFont = (asciiLut == main_font_lut);
-
     asciiLut = segmented_to_virtual(asciiLut);
 
-    while ((c = str[strPos]) != '\0' && c != '\n') {
+    while ((c = str[++strPos]) != '\0' && c != '\n') {
         // Handle color codes and tabs if using generic font
         if (isGenericFont) {
-            if (c == CHAR_COLOR_CODE) {
-                if (is_color_code_valid(str, strPos + 1)) {
-                    strPos += 7;
-                    continue;
-                }
-            } else if (c == '\t') {
-                width += 4 * SPACE_KERNING(asciiLut);
-                strPos++;
-                continue;
+            switch (c) {
+                case '\t':
+                    width += 4 * SPACE_KERNING(asciiLut);
+                    break;
+                case HEX(CONTROL_CHAR_COL):
+                    if (is_color_code_valid(str, strPos + 1)) {
+                        strPos += sizeof(gDialogCarryoverColor) * 2;
+                    }
+                    break;
+                case HEX(CONTROL_CHAR_RESET):
+                    break;
+                default:
+                    goto standard_character;
+                    break;
             }
         }
+
+standard_character:
```

Comment:
don't use a goto/switch here, just an if else if chain is fine. the switch won't compile to a jump table anyway